feature-libs/pickup-in-store/components/container/store-list/store-list.component.ts

```diff
@@ -43,16 +54,49 @@ export class StoreListComponent implements OnInit {
       this.productCode
     );
     this.isSearchRunning$ = this.pickupLocationsSearchService.isSearchRunning();
+
+    this.activeCartFacade
+      .getActive()
+      .pipe(
+        tap((cart) => {
+          this.cartId = cart.guid ?? '';
+          this.userId = cart.user?.uid ?? '';
+          cart.entries &&
+            cart.entries.forEach((entry) => {
+              if (entry.product?.code === this.productCode) {
+                this.entryNumber = entry.entryNumber ?? -1;
+                this.quantity = entry.quantity as number;
+              }
+            });
+        })
+      )
+      .subscribe();
   }
 
   onSelectStore(store: PointOfServiceStock) {
     const { stockInfo: _, ...pointOfService } = store;
-    const { name, displayName } = pointOfService;
+    const { name = '', displayName } = pointOfService;
+
+    const payload: SetPickupOptionInStorePayload = {
+      deliveryPointOfService: {
+        name,
+      },
+      quantity: this.quantity,
+    };
```

Comment:
Similarly to my previous comments, we should not build the payload in a component.